### DISCUSSION TITLE
Semester 4 Woche 8

### DISCUSSION THREAD

**Discussion by TimSchnur** [Christian-2003/CrInGE#49] [Student Exchange]
Hallo zusammen,

diese Woche haben wir uns zusammengesetzt und die Schnittstelle zwischen Editor und Engine 👁️ reviewed.
Einen ausführlichen [📝 Bericht über alle detaillierten Ergebnisse des Reviews](https://github.com/Christian-2003/CrInGE/blob/master/docs/ReviewMeetingProtokoll.md) findet ihr in unserem Docs Verzeichnis.
Die Zusammenfassung seht ihr allerdings auch ⬇️ unten.

Außerdem ist der [🧪 Testbericht](https://github.com/Christian-2003/CrInGE/blob/master/docs/Testbericht.md) jetzt fertig. Ihr findet diesen natürlich auch in unserem Docs Verzeichnis.

Das wäre dann auch schon alles für diese Woche. Bis zum nächsten Blog dauert es jetzt ein wenig länger, aber dann werden wir euch über einen hoffentlich großen Fortschritt des Projektes informieren.

## Ergebnisse des Meetings
Nach ausführlicher Betrachtung des *game_datahandler* wurden folgende Schlüsse gezogen:

### Abgeleitete Aufgaben
Da der Code seine funktionalen Anforderungen erfüllt und für die Demo immernoch einige Entwicklungsarbeit ansteht, werden die vorhandenen Ressourcen auf das Erstellen der Demo gelenkt und von einem erneuten Refactoring abgesehen.

### Bewährte Praktiken
Die funktionalen Anforderungen an das Package wurden erfüllt und fristgerecht abgegeben.  
Außerdem sind im Sinne der Vollständigkeit Funktionen Implementiert, welche momentan noch nicht genutzt werden, allerdings in Zukunft eine Rolle spielen können.

### Verbesserungspotenzial
Die JavaDoc Kommentare müssen nach jeder Änderung des Codes auf Korrektheit geprüft werden, um diese stehts aktuell zu halten.  
Außerdem soll die Standart Maven Projekt Strukur eingehalten werden.

Durch mehrere kleine Funktionen und Klassen, können große zyklomatische Komplexität und Modularität erreicht werden.  
Beim arbeiten mit Bäumen können diese mittels Rekursion traversiert werden, um die Wartbarkeit zu erhöhen.

**Comment by davidlzt** [comment]
Hallo zusammen,

toller Eintrag! Besonders beeindruckend fand ich die detaillierte Analyse und die klar formulierten abgeleiteten Aufgaben. Auch die Identifizierung von Verbesserungspotenzialen ist sehr hilfreich. Ich freue mich schon auf den nächsten Blog und bin gespannt auf die Fortschritte, die ihr dann präsentieren werdet.

Viele Grüße, XCalendar!
